eng/packages/http-client-csharp-mgmt/emitter/test/resolve-arm-resources-validation.test.ts

```diff
@@ -0,0 +1,1276 @@
+
+    // Verify Employee resource
+    strictEqual(
+      employeeResource.resourceInstancePath,
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContosoProviderHub/companies/{companyName}/departments/{departmentName}/employees/{employeeName}",
+      "Employee resourceInstancePath should match"
+    );
+    strictEqual(
+      getResourceTypeString(employeeResource),
+      "Microsoft.ContosoProviderHub/companies/departments/employees",
+      "Employee resource type should match"
+    );
+    strictEqual(
+      getResourceScope(employeeResource),
+      "ResourceGroup",
+      "Employee should be ResourceGroup scoped"
+    );
+
+    // Verify parent chain
+    ok(employeeResource.parent, "Employee should have a parent");
+    strictEqual(
+      employeeResource.parent.type.name,
+      "Department",
+      "Employee's parent should be Department"
+    );
+    ok(employeeResource.parent.parent, "Department should have a parent");
+    strictEqual(
+      employeeResource.parent.parent.type.name,
+      "Company",
+      "Department's parent should be Company"
+    );
+
+    // Verify Department resource
+    strictEqual(
+      departmentResource.resourceInstancePath,
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContosoProviderHub/companies/{companyName}/departments/{departmentName}",
+      "Department resourceInstancePath should match"
+    );
+    strictEqual(
+      getResourceTypeString(departmentResource),
+      "Microsoft.ContosoProviderHub/companies/departments",
+      "Department resource type should match"
+    );
+
+    // Verify Company resource
+    strictEqual(
+      companyResource.resourceInstancePath,
+      "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContosoProviderHub/companies/{companyName}",
+      "Company resourceInstancePath should match"
+    );
+    strictEqual(
+      getResourceTypeString(companyResource),
+      "Microsoft.ContosoProviderHub/companies",
+      "Company resource type should match"
+    );
+    strictEqual(companyResource.parent, undefined, "Company should have no parent");
+  });
+
+  it("resource with grand parent under a subscription", async () => {
+    const program = await typeSpecCompile(
+      `
+/** A Company grandparent resource */
+@subscriptionResource
+model Company is TrackedResource<CompanyProperties> {
+  ...ResourceNameParameter<Company>;
+}
+
+/** Company properties */
+model CompanyProperties {
+  /** Name of company */
+  name?: string;
+}
+
+/** A Department parent resource */
+@subscriptionResource
+@parentResource(Company)
+model Department is TrackedResource<DepartmentProperties> {
+  ...ResourceNameParameter<Department>;
+}
+
+/** Department properties */
+model DepartmentProperties {
+  /** Name of department */
+  name?: string;
+}
+
+/** An Employee resource with grandparent */
+@subscriptionResource
```

Comment:
Note that there is no need to mark any resource but the grandparent as a subscription resource

Reply:
make sense, we should remove this.